numpy/ma/core.pyi

```diff
@@ -992,7 +992,7 @@ diagonal: _frommethod
 harden_mask: _frommethod
 ids: _frommethod
 mean: _frommethod
-nonzero: _frommethod
+def nonzero(self: ArrayLike, /) -> tuple[NDArray[intp], ...]: ...
```

Comment:
https://github.com/numpy/numpy/pull/28794#discussion_r2056679520 also applies here

Reply:
yup thanks - OK if I do some work on the `MaskedArray` methods, which are typically much simpler, and then move over to the top-level ones once I've addressed https://github.com/numpy/numpy/pull/28794#discussion_r2056679520 ?

Reply:
sounds like a good plan 👌🏻